src/annotator/integrations/test/image-text-layer-test.js

```diff
@@ -0,0 +1,307 @@
+import { ImageTextLayer } from '../image-text-layer';
+
+// Sizes and spacing between character bounding boxes in these tests, expressed
+// as fractions of the image size.
+const charWidth = 0.04;
+const charHeight = 0.08;
+const charSpacing = 0.05;
+const lineSpacing = 0.1;
+
+/**
+ * Create character bounding box data for text in an image.
+ */
+function createCharBoxes(text) {
+  const charBoxes = [];
+  let lineIndex = 0;
+  let charIndex = 0;
+
+  for (let char of text) {
+    charBoxes.push({
+      left: charIndex * charSpacing,
+      right: charIndex * charSpacing + charWidth,
+      top: lineIndex * lineSpacing,
+      bottom: lineIndex * lineSpacing + charHeight,
+    });
+
+    if (char === '\n') {
```

Comment:
Maybe using `\r`, like VS does?

```suggestion
    if (char === '\r') {
```

Reply:
They do? I hadn't noticed that. I think we should stick with Unix-y conventions in general-purpose code but if VitalSource's OCR output uses `\r` then we'll have to handle that if new lines alter the text layer output in future - currently the text layer builder only distinguishes ASCII whitespace from other chars.